setup.py

```diff
@@ -90,6 +90,7 @@ def setup(*args, **kwargs):
         "scipy>=1.1",
     ],
     tests_require=[
+        "ase==3.21.1",
```

Comment:
Test pinnings in `setup.py` should be a minimum version (they do not affect our CI). If a version older than 3.21.1 works, then change the number.
```suggestion
        "ase>=3.21.1",
```

Reply:
Wouldn't this cause pip to fail to install when installing with tests due to https://github.com/glotzerlab/freud/pull/660#issuecomment-915550017  ?

Reply:
We don't install our testing dependencies in CI from `setup.py`. These values are just for user reference (or if the user requests to install the testing dependencies using some `python setup.py ...` command). That's also why these are all very broad pinnings like `>=MAJOR.0`. I identified that `3.16.0` was the lowest version we'll support, so I'm pinning this to `>=3.16`.